File: domaene/src/test/java/de/therapeutenkiller/haushaltsbuch/domaene/MoneyConverter.java
package de.therapeutenkiller.haushaltsbuch.domaene;

import cucumber.api.Transformer;
import org.apache.commons.lang3.StringUtils;

import javax.money.MonetaryAmount;

public class MoneyConverter extends Transformer<MonetaryAmount> {

    @Override
    public final MonetaryAmount transform(final String währungsbetrag) {
        if (StringUtils.isEmpty(währungsbetrag)) {
            throw new IllegalArgumentException("Währungsbetrag ist leer");
        }

        final WährungsbetragParsen parsen = new DeutschenWährungsbetragParsen();
        return parsen.aus(währungsbetrag);
    }
}

File: testit/src/test/kotlin/com/github/haschi/domain/haushaltsbuch/testing/MoneyConverter.kt
package com.github.haschi.domain.haushaltsbuch.testing

import cucumber.api.Transformer
import com.github.haschi.dominium.haushaltsbuch.core.model.values.Währungsbetrag
import com.github.haschi.dominium.haushaltsbuch.core.model.values.euro

class MoneyConverter : Transformer<Währungsbetrag>()
{

    override fun transform(währungsbetrag: String): Währungsbetrag
    {
        if (währungsbetrag.isEmpty())
            return 0.0.euro()

        return Währungsbetrag.währungsbetrag(währungsbetrag)
    }
}
